octobot/updater/python_updater.py

```diff
@@ -0,0 +1,123 @@
+#  Drakkar-Software OctoBot
+#  Copyright (c) Drakkar-Software, All rights reserved.
+#
+#  This library is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU Lesser General Public
+#  License as published by the Free Software Foundation; either
+#  version 3.0 of the License, or (at your option) any later version.
+#
+#  This library is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public
+#  License along with this library.
+import json
+import os
+import subprocess
+from shlex import quote as shlex_quote
+
+import aiohttp
+import sys
+from packaging.version import parse
+
+import octobot.constants as constants
+import octobot.updater.updater as updater_class
+
+
+class PythonUpdater(updater_class.Updater):
+    def __init__(self):
+        super().__init__()
+        self.use_git = os.path.isdir(".git")
+
+    async def get_latest_version(self):
+        if self.use_git:
+            try:
+                self._run_git_fetch()
+                return self._run_git_get_latest_tag()
+            except subprocess.CalledProcessError as e:
+                self.logger.debug(f"Failed to update with git : {e}")
+                return None
+        # with pip
+        return self._get_latest_pypi_version_from_data(await self._get_latest_pypi_version_data())
+
+    def _get_latest_pypi_release_url(self):
+        return f"https://pypi.python.org/pypi/{constants.PROJECT_NAME}/json"
+
+    async def _get_latest_pypi_version_data(self):
+        try:
+            async with aiohttp.ClientSession().get(self._get_latest_pypi_release_url()) as resp:
+                text = await resp.text()
+                if resp.status == 200:
+                    return json.loads(text)
+            return None
+        except Exception as e:
+            self.logger.debug(f"Error when fetching latest pypi package data : {e}")
+
+    def _get_latest_pypi_version_from_data(self, pypi_package_data):
+        try:
+            releases = pypi_package_data.get('releases', [])
+            version = parse('0')
```

Comment:
Not sure I understand the purpose of this line

Reply:
This is the initial version, used to be compared with max()

Reply:
Oh, right !